File: src/main/java/link/rdcn/user/exception/ClientException.java
package link.rdcn.user.exception;

import io.grpc.Metadata;
import io.grpc.Status;
import io.grpc.StatusRuntimeException;
import org.apache.arrow.flight.FlightRuntimeException;

public class ClientException extends Exception {
    private final ErrorCode errorCode;

    public ClientException(Exception e) {
        super();
        this.errorCode = ErrorCode.fromException(e);
    }

    public ClientException(FlightRuntimeException e) {
        super();
        this.errorCode = ErrorCode.fromException(e);
    }

    public ErrorCode getErrorCode() {
        return errorCode;
    }
}

File: faird-core/src/main/scala/link/rdcn/client/exception/ClientException.scala
package link.rdcn.client.exception

import link.rdcn.ErrorCode
import link.rdcn.util.ExceptionHandler

class ClientException(errorCode: ErrorCode) extends Exception {
  def getErrorCode: ErrorCode = errorCode
}

object ClientException {
  def apply(e: Exception): ClientException =
    new ClientException(ExceptionHandler.getErrorCode(e))
}